### DISCUSSION TITLE
YAML-шапка статей

### DISCUSSION THREAD

**Discussion by furtivite** [doka-guide/content#1163] [Q&A] [answered]
У нас есть техническая информация, которая хранится в YAML-формате в шапке статей. 

Она описана в примерах [для статьи](https://github.com/doka-guide/content/blob/main/docs/examples/article.md) и [для доки](https://github.com/doka-guide/content/blob/main/docs/examples/doka.md) (кажется, тут можно оставить один файл). 

Выглядит примерно так:

```markdown
---
title: "Название статьи"
authors:
  - Никнейм основного автора
contributors:
  - Никнеймы всех соавторов и контрибьюторов
designers: Никнейм дизайнера
summary:
  - Альтернативные теги для работы поиска
cover:
  desktop: 'images/desktop.png'
  mobile: 'images/mobile.png'
  alt: 'Альтернативное описание для обложки'
tags:
  - doka
---
```

На техническом созвоне 16 июля мы договорились:

- Что нужно определиться в порядке таких полей, потому что автоматически это проверять невозможно
- Что нужно понять какие поля нам могут понадобиться дополнительно
- Что нужно поле типа `starred` или `featured` - чтобы выводить статью на главную

И раньше мы договорились, что поле `summary` будет переименовано в `techs` или что-то вроде того. 

После обсуждения все поля надо будет привести к единому формату в #1056

**Comment by furtivite** [comment]
@pepelsbey, @tachisis, @solarrust, @nlopin, @igsekor, @monochromer приглашаются к обсуждению.

**Comment by furtivite** [comment]
Мне кажется, что большие поля-списки типа: ex-summary и tags — должны находиться где-то в конце списка, чтобы их править было удобнее.

**Comment by tachisis** [comment]
У меня особого мнения нет, как решите, так и ладно. Редакторский тег я обычно ставлю после всех авторов и контрибьюторов, так принято в книжном деле.

**Comment by igsekor** [comment]
Насколько я помню, мы договорились переименовать `summary` в `keywords`.

Последовательность полей, наверное, должна определяться редакторами из соображений безопасности удобства.

Мне со своей колокольни кажется логичным разделить поля на две группы: служебные и контентные (SEO?). Можно сделать примерно так:

- title
- description
- keywords
- tags
- cover
- authors
- contributors

Кажется, что в мете ещё должно появиться поле `description` для SEO.

**Comment by pepelsbey** [comment]
Я бы начал с тех полей, которые мы реально используем сейчас и определился с порядком:

```yml
title: 'Название статьи'
description: 'Описание статьи'
authors
  - author
contributors
  - contributor
editors
  - editor
keywords
  - слово
cover: 'Описание обложки'
```

Обратите внимание на описание обложки: если у нас есть поле со строкой-описанием, то из этого можно сделать много выводов:

- Что есть обложка
- Что у неё есть описание
- Что путь к обложке стандартные _cover/desktop.png_ и _cover/mobile.png_

Всё это поможет разгрузить шапку от лишнего. Плюс это работает похоже на поле `author`: мы же не пишем там путь к файлу автора, мы просто подразумеваем, что есть коллекция people, где мы будем искать этот ник. Так же и с обложкой.

Плюс стоит решить одиночные или двойные кавычки. Мне кажется, что лучше одиночные :)

**Comment by monochromer** [comment]
Мне больше нравится, когда сначала идут все поля, которые непосредственно относятся к контенту, затем люди, затем "бизнес"-поля:
```yaml
title
description
cover

authors
contributors
editors

keywords
tags
```

**Reply by pepelsbey** [reply]
Выглядит здорово, +1

**Comment by nlopin** [comment]
Мы это уже запилили #1172 или осталось что-то еще по этой теме? 

Если всё готово, то давайте напишем итог, пометим его как верный ответ и закроем топик

**Reply by solarrust** [reply]
Поддерживаю предыдущего оратора!
@pepelsbey

**Comment by pepelsbey** [comment, marked as answer]
У нас уже есть решение, оно описано во [frontmatter.json](https://github.com/doka-guide/content/blob/main/.frontmatter.json)
